### DISCUSSION TITLE
losing props inside a component when using ClientOnly and fallback

### DISCUSSION THREAD

**Discussion by sandros94** [nuxt/nuxt#20419] [Questions]
Hi, I'm still not sure if to open a issue or not, so I decided to open up a Discussion instead.

While working on a [test-repo](https://github.com/Sandros94/test-nuxt3app/tree/Anu) with Nuxt, UnoCSS and [Anu](https://github.com/jd-solanki/anu) I've noticed that some icons (parsed by UnoCSS from CDN) inside an Anu component (`ASwitch`) weren't loading once in production.
But when discussing about it in [this issue here](https://github.com/jd-solanki/anu/issues/155), I've noticed that it was not UnoCSS/Anu's fault, but rather that the props inside the `#fallback` would overwritten the ones of the actual `<ClientOnly>` component.

### What I was going for
I wanted to create a switch that would handle the switch between the system's theme and the whatever light/dark wasn't used by the system.

To do so I've used `nuxtjs/color-mode` and Anu's `ASwitch`:
```vue
<template>
    <ClientOnly>
        <template #fallback>
            <ASwitch v-model="off" off-icon="i-svg-spinners:clock"/>
        </template>
        <ASwitch v-model="switchTheme" off-icon="i-ph:sun" on-icon="i-ph:moon"/>
    </ClientOnly>
</template>

<script setup>
const colorMode = useColorMode();
const off = ref(false)
const switchTheme = computed({
    get: () => colorMode.value === 'dark',
    set: (value) => {
        if (colorMode.preference === 'system') {
            // Set the opposite of the current system value
            colorMode.preference = value ? 'dark' : 'light';
        } else {
            // Set the preference back to 'system'
            colorMode.preference = 'system';
        }
    }
});
</script>
```
### The need of `<ClientOnly>`
Since I wouldn't know what theme the browser is currently using, unless the page is loaded, to prevent a faulty state of the `ASwitch` component I decided to wrap it around a `<ClientOnly>`. And to make things prettier I've added a `#fallback`, so that the switch wouldn't appear from nowhere at the end of the page load.

### But once i run `npm run build`
for light theme:
![image](https://user-images.githubusercontent.com/13056429/233406405-603a7e22-4750-47d1-97f8-0b05c50e09e7.png)
for dark theme:
![image](https://user-images.githubusercontent.com/13056429/233406268-d7d95b91-b340-4d95-99d8-40860e8aa177.png)

### What I think I'm getting
Once the page is fully loaded, the `<ClientOnly>` with the `#fallback` instead of being rendered as:
```vue
<ASwitch v-model="switchTheme" off-icon="i-ph:sun" on-icon="i-ph:moon"/>
```
it gets rendered as:
```vue
<ASwitch v-model="switchTheme" off-icon="i-svg-spinners:clock"/>
```

Basically it correctly updates to the right `v-model` but it keeps the custom props of the `#fallback` component.

**Comment by sandros94** [comment]
How can I understand if this issue is related to Nuxt or Anu?
